src/crimson/net/SocketMessenger.cc

```diff
@@ -96,15 +94,13 @@ seastar::future<> SocketMessenger::start(Dispatcher *disp)
 
   // start listening if bind() was called
   if (listener) {
-    seastar::repeat([this] {
+    seastar::keep_doing([this] {
         return listener->accept()
           .then([this] (seastar::connected_socket socket,
                         seastar::socket_address paddr) {
             // start processing the connection
-            accept(std::move(socket), paddr)
+            return accept(std::move(socket), paddr)
```

Comment:
we dont want to return accept()'s future here - we want to start accepting the next connection immediately